sdk/upload.go

```diff
@@ -0,0 +1,186 @@
+package sdk
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"sync"
+
+	"github.com/root-gg/plik/server/common"
+)
+
+type PlikUpload = common.Upload
+
+// Upload override from a Plik common.Upload
+// to add some handy getters in SDK
+type Upload struct {
+	*PlikUpload
+	Files  map[string]*File `json:"files"`
+	client *Client
+}
+
+// UploadStatus
+type UploadStatus struct {
+	Error error
+	File  *File
+}
+
+// UploadOptions is a handy class to manage upload options from SDK
+// instead of being set in the common.Upload object
+type UploadOptions struct {
```

Comment:
Why not reuse struct in common/upload.go to avoid duplication and desynchronization between SDK and server ?

Reply:
I did this to avoid user to have to create an common.Upload and give it to NewUpload() that would also returns a common.Upload which is kind of confusing :/